shm/Cargo.cpp

```diff
@@ -0,0 +1,19 @@
+#include "Cargo.hpp"
+
+Cargo& Cargo::operator+=(const size_t amount) {
+    amount_ += amount;
+    return *this;
+}
+
+Cargo& Cargo::operator-=(const size_t amount) {
+    if (amount > amount_) {
+        amount_ = 0;
+        return *this;
+    }
+    amount_ -= amount;
+    return *this;
+}
+
+bool Cargo::operator==(const Cargo& cargo) {
+    return (name_ == cargo.name_);
```

Comment:
I think that if you are checking if cargo is the same, it will be worth to check also (at least) basePrice

Reply:
We decided to leave it as is for now because we didn't know how this would be used. If/when it will turn out that more thorough check is needed it can be implemented

Reply:
@PierwszyKrytykRP  Agree :)